tests/core/test_foundation.py

```
@@ -781,6 +781,86 @@ class Meta:
+def test_template_dirs_via_config_precedence(rando):
+    # config-supplied template_dirs replaces ``Meta.template_dirs``
+    # and sits in that same precedence slot — between
+    # ``core_system_template_dirs`` and ``template_dir`` /
+    # ``core_user_template_dirs``.
+    label = f"app-{rando}"
+    defaults = init_defaults(label)
+    defaults[label]['template_dirs'] = ['/cfg/dir']
+
+    class ThisTestApp(TestApp):
+        class Meta:
+            core_system_template_dirs = ['/sys/dir']
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)